test/integration/messages.js

```diff
@@ -0,0 +1,281 @@
+		ownerChannel.id,
+		{ created_by_id: ownerID },
+	);
+	const userChannel = userClient.channel(ownerChannel.type, ownerChannel.id);
+	if (createOnServerSide) {
+		await serverSideChannel.create();
+	} else {
+		await ownerChannel.create();
+	}
+	await serverSideChannel.addModerators([modID]);
+	return {
+		owner: {
+			id: ownerID,
+			client: ownerClient,
+			channel: ownerChannel,
+		},
+		member: {
+			id: memberID,
+			client: memberClient,
+			channel: memberChannel,
+		},
+		mod: {
+			id: modID,
+			client: modClient,
+			channel: modChannel,
+		},
+		user: {
+			id: userID,
+			client: await userClient,
+			channel: userChannel,
+		},
+		serverSide: {
+			client: serverSideClient,
+			channel: serverSideChannel,
+		},
+	};
+}
+
+function testMessagePinPermissions(type, createOnServerSide, matrix) {
+	let chat;
+	before(async () => {
+		chat = await setupTestChannel(type, createOnServerSide);
+	});
+	Object.entries(matrix).forEach((entry) => {
+		const [role, allowed] = entry;
+		if (allowed) {
+			it(role + ' is allowed to pin a message', async () => {
+				await chat[role].channel.sendMessage({
+					text: 'Pinned message',
+					pinned: true,
+				});
+			});
+		} else {
+			it(role + ' is not allowed to pin a message', async () => {
+				await expectHTTPErrorCode(
+					403,
+					chat[role].channel.sendMessage({
+						text: 'Pinned message',
+						pinned: true,
+					}),
+				);
+			});
+		}
+	});
+}
+
+describe('pinned messages', () => {
+	context('general use cases', () => {
+		let chat;
+		before(async () => {
+			chat = await setupTestChannel('messaging', false);
+		});
+		it('send pinned message', async () => {
+			await chat.owner.channel.sendMessage({
+				text: 'Regular message 1',
+				pinned: false,
+			});
+			const { message } = await chat.owner.channel.sendMessage({
+				text: 'Pinned message 1',
+				pinned: true,
+			});
+			expect(message.pinned).to.be.equal(true);
+			expect(message.pinned_at).to.be.not.empty;
+			expect(message.pinned_by.id).to.be.equal(chat.owner.id);
+
+			const { results } = await chat.owner.channel.search({ pinned: true });
+			expect(results.length).to.be.equal(1);
+			expect(results[0].message.id).to.be.equal(message.id);
+		});
+
+		it('pin message', async () => {
+			const { message } = await chat.owner.channel.sendMessage({
+				text: 'Regular message 2',
+			});
+			expect(message.pinned).to.be.equal(false);
+			const { message: updatedMessage } = await chat.owner.client.updateMessage({
+				id: message.id,
+				pinned: true,
+			});
+			expect(updatedMessage.pinned).to.be.equal(true);
+		});
+
+		it('unpin message', async () => {
+			const { message } = await chat.owner.channel.sendMessage({
+				text: 'Pinned message 2',
+				pinned: true,
+			});
+			expect(message.pinned).to.be.equal(true);
+			const { message: updatedMessage } = await chat.owner.client.updateMessage({
+				id: message.id,
+				pinned: false,
+			});
+			expect(updatedMessage.pinned).to.be.equal(false);
```

Comment:
can you check if the message text persists after unpin?

Reply:
time to add partial message update :)

Reply:
makes sense actually :sweat_smile:

Reply:
Yup, missed that. I've fixed the tests, but the partial update would be a topic for discussion in a different PR I think